bin/conform-templates.php

```diff
@@ -0,0 +1,299 @@
+    unset($content['type']);
+}
+$content['license'] = $templateContent['license'];
+$content['support'] = $templateContent['support'];
+if (! $hasDocs) {
+    unset($content['support']['docs']);
+}
+foreach ($content['support'] as &$supportLink) {
+    $supportLink = strtr($supportLink, $replace);
+}
+
+$content['config'] = $templateContent['config'];
+$content['scripts'] = $templateContent['scripts'];
+
+if (empty($content['keywords'])) {
+    fwrite(STDERR, 'Missing "keywords" in composer.json');
+} else {
+    $hasZf = false;
+    $hasZendframework = false;
+    foreach ($content['keywords'] as &$keyword) {
+        if ($keyword === 'zf2') {
+            $keyword = 'zf';
+            $hasZf = true;
+        } elseif ($keyword === 'zf') {
+            $hasZf = 'zf';
+        } elseif ($keyword === 'zendframework') {
+            $hasZendframework = true;
+        }
+    }
+
+    if (! $hasZendframework) {
+        array_unshift($content['keywords'], 'zendframework');
+    }
+    if (! $hasZf) {
+        array_unshift($content['keywords'], 'zf');
+    }
+}
+
+if (file_exists('.docheader')) {
+    array_unshift($content['scripts']['check'], '@license-check');
+    $content['scripts']['license-check'] = 'docheader check src/ test/';
+}
+
+unset($content['minimum-stability'], $content['prefer-stable']);
+
+$list = json_decode(
+    file_get_contents('https://docs.zendframework.com/zf-mkdoc-theme/scripts/zf-component-list.json'),
+    true
+);
+
+$description = null;
+foreach ($list as $component) {
+    if (strpos($component['url'], '/' . $repo . '/') !== false) {
+        $description = rtrim($component['description'], '.');
+        break;
+    }
+}
+
+if ($description !== null) {
+    $content['description'] = $description;
+}
+
+// sort section in composer:
+
+uksort($content, function ($a, $b) use ($sectionOrder) {
+    $ia = array_search($a, $sectionOrder);
+    $ib = array_search($b, $sectionOrder);
+
+    if ($ia === $ib) {
+        return 0;
+    }
+
+    if ($ia === false) {
+        return 1;
+    }
+
+    if ($ib === false) {
+        return -1;
+    }
+
+    if ($ia < $ib) {
+        return -1;
+    }
+
+    return 1;
+});
+
+file_put_contents(
+    'composer.json',
+    json_encode(
+            $content,
+            JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES
+    ) . PHP_EOL
+);
+
+// update year in mkdocs.yml
+if ($hasDocs) {
+    $content = file_get_contents('mkdocs.yml');
+    $content = preg_replace('/docs_dir:.*/', 'docs_dir: docs/book', $content);
+    $content = preg_replace('/site_dir:.*/', 'site_dir: docs/html', $content);
+
+    if (preg_match('/Copyright \(c\) (\d{4})/', $content, $m)) {
+        $year = $m[1];
+        $copyrightYear = $year < $currentYear ? $year . '-' . $currentYear : $currentYear;
+        $content = preg_replace(
+            '/Copyright \(c\) \d{4}(-\d{4}) /',
+            'Copyright (c) ' . $copyrightYear . ' ',
+            $content
+        );
+        file_put_contents('mkdocs.yml', $content);
```

Comment:
We want to rewrite the contents regardless of whether or not we were able to update the copyright date; this should be moved outside the conditional.

Reply:
Good point, fixed in af5c781